link.py

```diff
@@ -0,0 +1,115 @@
+from collections import deque
+'''
+   Class for a link.
+   
+   Attributes:
+   link_id is a unique identifier for each link
+   max_buffer_size is how much data can be stored in the link buffer
+   delay is propogation delay of the link
+   capacity is the maximum link rate
+   nodes is an array of the 2 nodes connected with this link
+   
+   link_buffer stores packets that are waiting to be sent on this link
+   
+   current_packet stores the packet being sent on the link.
+   current_destination stores which host this packet is being sent to.
+   '''   
+
+class Link(object):
+    '''
+    Constructor for Link class takes the aformentioned 
+    link_id,
+    max_buffer_size,
+    delay,
+    capacity,
+    nodes'''    
+    def __init__ (self, link_id, max_buffer_size, delay, capacity, nodes):
```

Comment:
Needs to take in instance of NetworkSimulator. Don't forget to add this to documentation as well.